### ISSUE TITLE
referrerPath not storing the correct value in session storage

### ISSUE THREAD

**Issue by baljinder-inmarsat** [okta/okta-oidc-js#786]
When I load my portal with the following URL e.g. www.mywebsite.com/site1/page1 and the login into okta. What happens is I get redirected back to  www.mywebsite.com/page1. You would expect to be redirect to what you initial had in your address bar.
 I've been debugging it and noticed that, in this example, Site1 gets removed in the sessionStorage 'referrerPath . Due to when you are setting the FromUri you are using window.location.origiin + fromUri. 

Suggestion: Just store the Path.

## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [X] okta-angular
- [ ] oidc-middleware
- [ ] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [X] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior

1. start portal with 'www.mywebsite.com/**site1**/page1'
2. Get sent to okta login page and login
3. get redirect to 'www.mywebsite.com/page1' 

## Expected behavior

1. start portal with 'www.mywebsite.com/**site1**/page1'
2. Get sent to okta login page and login
3. get redirect to 'www.mywebsite.com/**site1**/page1'

## Minimal reproduction of the problem with instructions
1. start portal with 'www.mywebsite.com/**site1**/page1'
2. Get sent to okta login page and login
3. get redirect to 'www.mywebsite.com/page1' 
## Environment

- Package Version:  "@okta/okta-angular": "^2.0.0",
- Browser: all
- OS: linux / window
- Node version (`node -v`): 12.4
- Other: angular 9

**Comment by mraible** [comment]
If you change your redirectUri to contain "/site1", everything should work as expected. `window.location.origin` does not include the path.

**Comment by baljinder-inmarsat** [comment]
I cannot find of a way to do  that  on a initial login, I can as a logout feature. As far as I can tell the library handles that via the OktaAuthGuard. Unless you are referring the redirectUrl in the config but that needs to be set with 'implicit/callback'

**Comment by mraible** [comment]
@baljinder-inmarsat Yes, I'm referring to the redirectUri in the config. The whole purpose of it is to tell Okta where you want your app to come back to. The only reason we use `window.location.origin` in our examples is because it's dynamic so you won't to change it in production. However, you will need to add the contextPath if your app has one.

**Comment by baljinder-inmarsat** [comment]
@mraible Sorry I think I didn't put the steps clearly enough, apologies. I get redirect back to my portal. 

1. start portal with 'www.mywebsite.com/site1/page1'
2. Get sent to okta login page and login
3. get redirect to 'www.mywebsite.com/site1/implicit/callback'
4. it hits handle authentication, does what it does. get sessionstorage.
5. Then redirects me to 'www.mywebsite.com/page1'

**Comment by aarongranick-okta** [comment]
@baljinder-inmarsat It seems you are serving your application with a base path of `/site1`. Do you have a `<base href=>` set in the document? I am also guessing that all routes are defined relative to the root,`site1`. Is this correct? Would you mind sharing your routing config? Specifically I am interested in how you are configuring angular so that it is aware of a root base url outside your application.

`okta-angular` is saving the URI here: https://github.com/okta/okta-oidc-js/blob/master/packages/okta-angular/src/okta/okta.guard.ts#L48

As a workaround, you could define an `onAuthRequired` function: https://github.com/okta/okta-oidc-js/tree/master/packages/okta-angular#okta_config here you can read the saved URI using `getFromUri`, convert it to an absolute URI and save it back using `setFromUri`

**Comment by baljinder-inmarsat** [comment]
Do you have a <base href=> set in the document?
index.html - <base href="/">
angular.json (depends on which environment ) - 
  "baseHref": "/site1/",
 "deployUrl": "/site1/",
Are the routes defined relative? Yes
Our Route Cong? I have  slightly tweaked to minimize it, plus the path values are in a constant file. I have add comments beside them to give the value they have been set to:
export const routes: Routes = [
  // Error page ROUTE
  // Stays out so that we can hide header and menubar
  {
    path: ROUTES_CONSTANTS.ERROR,  // error/
    component: ErrorPageComponent,
    data: {
      title: 'ERRORS.TITLE'
    }
  },

  {
    path: '',
    component: LayoutComponent,
    canActivateChild: [InmOktaAuthGuardService, AuthGuardService], // TODO lets replace InmOktaAuthGuardService with OktaAuthGuard when they implement it
    children: [
      // APP ROUTES_CONSTANTS
      {
        path: ROUTES_CONSTANTS.DASHBOARD,  // we have a constant file, value is: ''
        loadChildren: () => import('./dashboard/dashboard.module').then(m => m.DashboardModule)
      },
  {
        path: ROUTES_CONSTANTS.ORDERS, // we have a constant file, value is: 'order'
        loadChildren: () => import('./orders/orders.module').then(m => m.OrdersModule),
        data: {
          access: [ xxx,yyy
          ]
        }
      }
      }
    ]
  },

  {
    path: '**',
    redirectTo: `/${ROUTES_CONSTANTS.NOT_FOUND}`
  }
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, {
      onSameUrlNavigation: 'reload',
      relativeLinkResolution: 'corrected',
      scrollPositionRestoration: 'top'
    })
  ],
  exports: [RouterModule]
})
export class AppRoutingModule {}

As a workaround, you could define an onAuthRequired?
That will involve adding code in my environment files, not too confident about that.

To me the issue is in the handle authentication, so my alternative is bit of hacky way. may give a idea of whats seems to be going wrong for me:

 .handleAuthentication()
      .then(() => {

        const postLogoutUrl = environment.OKTA.postLogoutRedirectUri;  // e.g www.website.com/site1
        const siteDomain = `${window.location.origin}/`;  // e.g www.website.com
        let fromUri = okta.getFromUri(); // e,g www.website.com/orders , site1 disappeared
        const isCorrectPath = fromUri.includes(postLogoutUrl); // check if site1 is there, which is not on initial login
        fromUri = isCorrectPath ? fromUri : fromUri.replace(siteDomain, postLogoutUrl);

        window.location.replace(fromUri);
      })

@aarongranick-okta hope this helps, thanks for checking this out!

**Comment by swiftone** [comment]
Internal ref: OKTA-306430
